src/app/api/tasks/tasks.service.ts

```diff
@@ -0,0 +1,96 @@
+import { CreateTaskRequest, UpdateTaskRequest } from '@/types/dto/tasks.dto'
+import { BaseService } from '@api/core/services/base.service'
+import { Resource } from '@api/core/types/api'
+import { PoliciesService } from '@api/core/services/policies.service'
+import { AssigneeType } from '@prisma/client'
+
+type FilterByAssigneeId = {
+  assigneeId: string
+  assigneeType: AssigneeType
+}
+
+export class TasksService extends BaseService {
+  private buildReadFilters() {
+    const user = this.user
+
+    let filters = { where: { workspaceId: user.workspaceId, OR: undefined as unknown as FilterByAssigneeId[] } }
+
+    if (user.clientId) {
+      filters = {
+        where: {
+          ...filters.where,
+          OR: [{ assigneeId: user.clientId as string, assigneeType: 'client' }],
```

Comment:
is this saying `assignedID = clientID OR assigneeType = client`? Because I don't think that is correct.

Reply:
The SQL returned by this prisma query is more along the lines of this:

```sql
SELECT ...
WHERE <prev conditions> AND (
    (`assigneeId` = 'clientId' AND `assigneeType` = 'client')
  )
```

When OR  has multiple values inside the array, it performs an OR - but if it only has a single value in its filter array it acts like a normal `AND` statement, without `OR` actually being involved.

The reason I wrote it in a slightly more non-conventional way is that I could override it later in `if (user.clientId && user.companyId)`, it seemed cleaner and shorter to overwrite the filter this way.